doc/exts/pylint_options.py

```diff
@@ -0,0 +1,190 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Script used to generate the options page."""
+
+from __future__ import annotations
+
+import re
+from collections import defaultdict
+from inspect import getmodule
+from pathlib import Path
+from typing import Dict, List, NamedTuple
+
+import tomlkit
+from sphinx.application import Sphinx
+
+from pylint.checkers import initialize as initialize_checkers
+from pylint.checkers.base_checker import BaseChecker
+from pylint.extensions import initialize as initialize_extensions
+from pylint.lint import PyLinter
+from pylint.typing import OptionDict
+from pylint.utils import get_rst_title
+
+
+class OptionsData(NamedTuple):
+    name: str
+    optdict: OptionDict
+    checker: BaseChecker
+    extension: bool
+
+
+OptionsDataDict = Dict[str, List[OptionsData]]
+
+PYLINT_BASE_PATH = Path(__file__).resolve().parent.parent.parent
+"""Base path to the project folder."""
+
+PYLINT_USERGUIDE_PATH = PYLINT_BASE_PATH / "doc" / "user_guide"
+"""Path to the messages documentation folder."""
+
+
+def _register_all_checkers_and_extensions(linter: PyLinter) -> None:
+    """Registers all checkers and extensions found in the default folders."""
+    initialize_checkers(linter)
+    initialize_extensions(linter)
+
+
+def _get_all_options(linter: PyLinter) -> OptionsDataDict:
+    """Get all options registered to a linter and return the data."""
+    all_options: OptionsDataDict = defaultdict(list)
+    for checker in sorted(
+        linter.get_checkers(),
+        key=lambda x: (
+            x.name != "master",
+            getmodule(x).__name__.startswith("pylint.extensions."),  # type: ignore[union-attr]
+            x.name,
+        ),
+    ):
+        ch_name = checker.name
+        for option in checker.options:
+            all_options[ch_name].append(
+                OptionsData(
+                    option[0],
+                    option[1],
+                    checker,
+                    getmodule(checker).__name__.startswith("pylint.extensions."),  # type: ignore[union-attr]
+                )
+            )
+
+    return all_options
+
+
+def _create_checker_section(
+    checker: str, options: list[OptionsData], linter: PyLinter
+) -> str:
+    checker_string = get_rst_title(f"``{checker.capitalize()}`` Checker", "^")
+
+    toml_doc = tomlkit.document()
+    pylint_tool_table = tomlkit.table(is_super_table=True)
+    toml_doc.add(tomlkit.key(["tool", "pylint"]), pylint_tool_table)
+
+    checker_table = tomlkit.table()
+
+    for option in sorted(options, key=lambda x: x):
```

Comment:
Isn't this the default ?
```suggestion
    for option in sorted(options):
```

Reply:
Oops, that should be `x.name` I think. Let me double-check.